app/models/grda_warehouse/client_pii.rb

```diff
@@ -0,0 +1,66 @@
+# mask PII attributes on the client record
+class GrdaWarehouse::ClientPii
+  attr_reader :user, :client
+
+  def initialize(user:, client:)
+    @user = user
+    @client = client
+  end
+
+  def first_name
+    name_part(client.first_name.presence)
+  end
+
+  def last_name
+    name_part(client.last_name.presence)
+  end
+
+  def middle_name
+    name_part(client.middle_name.presence)
+  end
```

Comment:
I think these should access the name part with the hash key format (`client[:FirstName]`) this would allow us to pass in a "client" that is really a hash so we can use this for situations where we've plucked the data and don't actually have an instance of `Client`.

The same would need to happen for where it calls `client.ssn` and `client.dob`. 

Alternatively we could new up a struct or open struct and pass it in with the appropriately named attributes, but that feels like it will be a bigger list and use more memory.

Reply:
I opted for to wrap attributes up in a struct. This does add some small memory overhead but makes the class cleaner imo. Could revisit if there's a need for micro-optimizations around memory usage